MapboxDirections/MBRouteOptions.swift

```diff
@@ -356,15 +356,23 @@ open class RouteOptions: NSObject, NSSecureCoding {
     open var includesExitRoundaboutManeuver = false
 
     /**
-     The locale in which the routes’ instructions are written.
+     The locale in which the route’s instructions are written.
 
      If you use MapboxDirections.swift with the Mapbox Directions API, this property affects the sentence contained within the `RouteStep.instructions` property, but it does not affect any road names contained in that property or other properties such as `RouteStep.name`.
 
      The Directions API can provide instructions in [a number of languages](https://www.mapbox.com/api-documentation/#instructions-languages). Set this property to `Bundle.main.preferredLocalizations.first` or `Locale.autoupdatingCurrent` to match the application’s language or the system language, respectively.
 
      By default, this property is set to `nil`, causing instructions to be written in the default language, English.
      */
-    open var locale: Locale?
+    open var locale: Locale? {
+        didSet {
+            guard let locale = locale else {
+                self.distanceMeasurementSystem = Locale.current.usesMetric ? .metric : .imperial
+                return
+            }
+            self.distanceMeasurementSystem = locale.usesMetric ? .metric : .imperial
```

Comment:
This is pretty cool, although are we sure it works in Objective-C? Maybe add some tests to that effect.

A simpler alternative would be to eschew these `didSet`s and move this logic to the `params` getter.

Reply:
Tested in the example: 

```
options.includesVoiceInstructions = YES;
options.locale = [NSLocale currentLocale];
```

And put a breakpoint on the next line to see what the value:

```
(lldb) po options.distanceMeasurementSystem
<nil>
```

Reply:
Hm, exactly the same changes in the Obj-C example:
```
(lldb) po options.distanceMeasurementSystem
MBMeasurementSystemImperial
```

Reply:
Cleaned my workspace and tried again, worked.